### ISSUE TITLE
`useless_attribute` does not permit `module_name_repetitions` on `use` statements

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/rust-clippy#9401]
### Summary

With `#![warn(clippy::module_name_repetitions)]`, clippy will emit a lint warning when `pub use` items start with the same prefix as the module with the `pub use` statement. This lint can be suppressed by adding an `allow` pragma on the `use` statement. However, once adding the allow, clippy emits a deny-by-default lint for `useless_attribute`.

`useless_attribute` does not permit `module_name_repetitions` on `use` items per its docs: https://rust-lang.github.io/rust-clippy/master/index.html#useless_attribute. I believe `useless_attribute` should permit `module_name_repetitions` on `use` items.

### Lint Name

useless_attribute

### Reproducer

I tried this code:

```rust
#![warn(clippy::module_name_repetitions)]

pub mod convert {
    mod internal {
        pub enum ConvertOnError {
            Yes,
            No,
        }
        
        pub fn convert_type() {}
    }
    
    pub use internal::{ConvertOnError, convert_type};
}

fn main() {
    println!("Hello, world!");
}
```

I saw this happen:

```
    Checking playground v0.0.1 (/playground)
warning: item name starts with its containing module's name
  --> src/main.rs:13:24
   |
13 |     pub use internal::{ConvertOnError, convert_type};
   |                        ^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src/main.rs:1:9
   |
1  | #![warn(clippy::module_name_repetitions)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_name_repetitions

warning: item name starts with its containing module's name
  --> src/main.rs:13:40
   |
13 |     pub use internal::{ConvertOnError, convert_type};
   |                                        ^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_name_repetitions

warning: `playground` (bin "playground") generated 2 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.58s
```

I tried to suppress the lint:

```rust
#![warn(clippy::module_name_repetitions)]

pub mod convert {
    mod internal {
        pub enum ConvertOnError {
            Yes,
            No,
        }
        
        pub fn convert_type() {}
    }
    
    #[allow(clippy::module_name_repetitions)]
    pub use internal::{ConvertOnError, convert_type};
}

fn main() {
    println!("Hello, world!");
}
```

and got this output:

```
    Checking playground v0.0.1 (/playground)
error: useless lint attribute
  --> src/main.rs:13:5
   |
13 |     #[allow(clippy::module_name_repetitions)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if you just forgot a `!`, use: `#![allow(clippy::module_name_repetitions)]`
   |
   = note: `#[deny(clippy::useless_attribute)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_attribute

error: could not compile `playground` due to previous error
```

Adding an additional suppression for `useless_attribute` results in the expected behavior:

```rust
#![warn(clippy::module_name_repetitions)]

pub mod convert {
    mod internal {
        pub enum ConvertOnError {
            Yes,
            No,
        }
        
        pub fn convert_type() {}
    }
    
    #[allow(clippy::module_name_repetitions)]
    #[allow(clippy::useless_attribute)] // clippy does not permit `clippy::module_name_repetitions` on `use` statements
    pub use internal::{ConvertOnError, convert_type};
}

fn main() {
    println!("Hello, world!");
}
```

and got this output: _no warnings_

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-apple-darwin
release: 1.63.0
LLVM version: 14.0.5
```
